instrumentation/netty/netty-4.1/javaagent/src/test/groovy/Netty41ClientTest.groovy

```diff
@@ -39,31 +42,50 @@ import spock.lang.Unroll
 class Netty41ClientTest extends HttpClientTest<DefaultFullHttpRequest> implements AgentTestTrait {
 
   @Shared
-  private Bootstrap bootstrap
+  private EventLoopGroup eventLoopGroup = buildEventLoopGroup()
 
-  def setupSpec() {
-    EventLoopGroup group = getEventLoopGroup()
-    bootstrap = new Bootstrap()
-    bootstrap.group(group)
+  @Shared
+  private Bootstrap bootstrap = buildBootstrap(false)
+
+  @Shared
+  private Bootstrap httpsBootstrap = buildBootstrap(true)
+
+  def cleanupSpec() {
+    eventLoopGroup?.shutdownGracefully()
+  }
+
+  Bootstrap buildBootstrap(boolean https) {
+    Bootstrap bootstrap = new Bootstrap()
+    bootstrap.group(eventLoopGroup)
       .channel(getChannelClass())
       .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, CONNECT_TIMEOUT_MS)
       .handler(new ChannelInitializer<SocketChannel>() {
         @Override
         protected void initChannel(SocketChannel socketChannel) throws Exception {
           ChannelPipeline pipeline = socketChannel.pipeline()
+          if (https) {
+            SslContext sslContext = SslContextBuilder.forClient().trustManager(InsecureTrustManagerFactory.INSTANCE).build()
```

Comment:
If possible we should avoid the insecure trust manager in tests too. Is the system property trustStore not enough here? If so it would be nice to read it and set up the trust manager manually using it.

Reply:
For some reason it was needed when we used google for https test. I didn't try without it after google -> localhost change.